src/test/kotlin/com/atlassian/performance/tools/report/chart/WaterfallChartTest.kt

```diff
@@ -100,4 +104,30 @@ class WaterfallChartTest {
             .`as`("should not throw when resource URL contains data URI scheme")
             .doesNotThrowAnyException()
     }
+
+    @Test
+    fun shouldDoNothingAndNotThrowWhenNavigationsAndResourcesAreEmpty() {
+        //given
+        val outputFile = Paths.get("build/i-should-not-exist.html").toFile()
+        val inputMetricsResource = "action-metrics-with-element-timing-only.jpt"
+        val actionMetrics: List<ActionMetric> =
+            ActionMetricsParser().parse(javaClass.getResourceAsStream(inputMetricsResource))
+        val metric = actionMetrics[0]
+
+        // when
+        val noExceptions = Assertions.catchThrowable {
+            WaterfallChart().plot(
+                metric = metric,
+                output = outputFile
+            )
+        }
+
+        // then
+        assertThat(noExceptions)
+            .`as`("should not throw when navigations and resources are empty")
+            .doesNotThrowAnyException()
+        assertThat(outputFile.exists())
+            .`as`("should not create any output")
+            .isFalse()
```

Comment:
Loving the explicit assertions :heart:

Reply:
It's my first contact with AssertJ and I'm really digging it!